snappi_ixnetwork/trafficitem.py

```diff
@@ -191,9 +220,245 @@ class TrafficItem(CustomField):
+            # after a while error is thrown. but could see the configuration
+            # applied at Ixnetwork. (Need to check with Eng team)
+            self._api._request("POST", url=url, payload=payload)
+        except Exception:
+            return
+        return
+
+    def get_ports_xpath(self, config):
+        ixn = self._api.assistant._ixnetwork
+        myfilter = [{"property": "name", "regex": ".*"}]
+        url, payload = self._get_search_payload(
+            "/",
+            "(?i)^(vport)$",
+            ["name"],
+            myfilter,
+        )
+        result = ixn._connection._execute(url, payload)
+        vports = {}
+        for vp in result:
+            if vp.get("vport") is None:
+                continue
+            for v in vp["vport"]:
+                vports[v["name"]] = v["xpath"]
+        return vports
+
+    def get_device_xpath(self, config):
+        if len(config.devices) == 0:
+            return {}
+        dev_names = []
+        for f in config.flows:
+            if f.tx_rx.choice == "port":
+                continue
+            dev_names.extend(f.tx_rx.device.tx_names)
+            dev_names.extend(f.tx_rx.device.rx_names)
+        dev_names = list(set(dev_names))
+        if dev_names == []:
+            return {}
+        selects = []
+        url = None
+        for name in dev_names:
+            href = self._api.get_ixn_href(name)
+            if href is None:
+                # TODO change to more detailed message
+                raise Exception(
+                    "href is not found for device name {}".format(name)
+                )
+            href = href.split("ixnetwork")[-1]
+            url, search = self._get_search_payload(
+                href, "^(xpath|ethernet|ipv4|ipv6|bgpv4|bgpv6)$", ["name"], []
+            )
+            selects.extend(search["selects"])
+        payload = {"selects": selects}
+        ixn = self._api.assistant._ixnetwork
+        result = ixn._connection._execute(url, payload)
+        if len(dev_names) != len(result):
+            # TODO change to more detailed message
+            raise Exception("search query result miss match with device names")
+        paths = {}
+        for i, d in enumerate(dev_names):
+            paths[d] = {"xpath": result[i]["xpath"]}
+            string = str(result[i])
+            if "ipv4" in string:
+                tr_type = "ipv4"
+            elif "ipv6" in string:
+                tr_type = "ipv6"
+            else:
+                tr_type = "ethernetVlan"
+            paths[d]["type"] = tr_type
+        return paths
+
+    def get_ixn_config(self, config):
+        ixn = self._api.assistant._ixnetwork
+        myfilter = [{"property": "name", "regex": ".*"}]
+        url, payload = self._get_search_payload(
+            "/traffic",
+            "(?i)^(trafficItem|configElement|frameRate"
+            "|frameSize|transmissionControl|stack|field|highLevelStream"
```

Comment:
probably stack|field are not require. As those are adding more payload. We will ignore those.

Reply:
traffic.py happens in 2 steps.
1. create the flows.
2. update the parameters of flows (exp: size, rate, duration, stack fields)
this will give the structure, will update the key value pairs.